src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -107,7 +107,8 @@ public StepExecution start(StepContext context) throws Exception {
             Map<String,String> overrides = new HashMap<String,String>();
             List<MultiBinding.Unbinder> unbinders = new ArrayList<MultiBinding.Unbinder>();
             for (MultiBinding<?> binding : step.bindings) {
-                if (binding.getDescriptor().requiresWorkspace() && workspace == null) {
+                if (binding.getDescriptor().requiresWorkspace() &&
+                        (workspace == null || launcher == null)) {
                     throw new MissingContextVariableException(FilePath.class);
```

Comment:
Misleading in case `FilePath` but not `Launcher` is present; again this is only theoretical.